source/material_model/rheology/strain_dependent.cc

```diff
@@ -283,6 +301,24 @@ namespace aspect
         friction_strain_weakening_factors = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Friction strain weakening factors"))),
                                                                                     n_fields,
                                                                                     "Friction strain weakening factors");
+
+        if (prm.get ("Strain healing mechanism") == "no healing")
+          healing_mechanism = no_healing;
+        else if (prm.get ("Strain healing mechanism") == "temperature dependent")
+          healing_mechanism = temperature_dependent;
+        else
+          AssertThrow(false, ExcMessage("Not a valid Strain healing mechanism!"));
+        
+        // Currently this functionality only works in field composition
+        if (healing_mechanism != no_healing)
+        {
+          AssertThrow(this->introspection().compositional_name_exists("particles"),
+                          ExcMessage("This healing mechanism currently does not work with particles!"));
+        }
```

Comment:
I think the line above is checking wether there is compositional field named `particles` and not whether the particles are being used as the "compositional field method". We can work on this after addressing other items.